### PR TITLE
[OPIK-2856] [BE] Add UUIDv7 time-based filtering for trace threads

### PR DESCRIPTION
**Pull request by thiagohora**
## Details

This PR implements UUIDv7 time-based filtering for trace threads, mirroring the functionality already implemented for traces in PR #3921 and spans in PR #3938.

### Key Changes

**Backend Implementation:**
- Added `fromTime` and `toTime` query parameters to `/api/v1/private/traces/threads` (paginated endpoint)
- Added `fromTime` and `toTime` fields to `/api/v1/private/traces/threads/search` (stream endpoint)
- Updated `TraceThreadCriteria` to include `uuidFromTime` and `uuidToTime` fields
- Modified `TraceDAO` SQL queries to filter trace threads by `thread_model_id` using UUID time bounds
- Updated `TracesResource` to validate and map time parameters using `InstantToUUIDMapper`

**Thread Model ID Generation (Critical Change):**
- **Previously**: Thread model IDs were generated using `Instant.now()`, meaning they reflected when the thread entry was created in the database
- **Now**: Thread model IDs are generated using the timestamp from the **earliest trace** in the thread
- Modified `TraceThreadListener` to track both the minimum trace ID (earliest timestamp) and the most recent `lastUpdatedAt` for each thread via a new `ThreadTimestamps` record
- Updated `TraceThreadIdService.getOrCreateTraceThreadId()` to accept an optional `Instant timestamp` parameter
- Modified `TraceThreadService` to extract the timestamp from the minimum trace ID using `IdGenerator.extractTimestampFromUUIDv7()`
- Added `IdGenerator.extractTimestampFromUUIDv7()` utility method to extract timestamps from UUIDv7
- Added `IdGenerator.generateId(Instant timestamp)` method to generate UUIDv7 from a specific timestamp
- **Result**: Time-based filtering on `thread_model_id` now accurately reflects the actual time range of traces within the thread, rather than when the thread entry was persisted

**Test Implementation:**
- Added comprehensive test suite `FindTraceThreadsTimeFilteringTests` with 10 test cases
- Tests cover both paginated and stream endpoints with parameterized tests
- Includes boundary condition testing, ISO-8601 format parsing, and validation scenarios
- Refactored tests to use helper methods (`createAndCloseThreads`, `assertTimeFilteredThreads`) to reduce ~120 lines of duplicated code
- Updated `TraceResourceClient` with new methods (`callGetTraceThreadsWithQueryParams`, `searchTraceThreadsStream`) for testing time-filtered endpoints
- Added 500ms delays after trace creation and thread closure to allow for asynchronous event processing

### Design Decisions

1. **Time filter on `thread_model_id`**: Applied to the `trace_threads` table ID column, which is now a UUIDv7 containing the timestamp of the earliest trace in the thread (not the thread creation time)
2. **Consistent with existing patterns**: Follows the same validation and mapping approach used for trace and span time filtering
3. **Both endpoints supported**: Both paginated (`/threads`) and stream (`/threads/search`) endpoints support the same time filtering parameters
4. **Backward compatible**: Existing threads without the timestamp-based ID generation continue to work; only new threads use the earliest trace timestamp

## Change checklist
- [x] User facing
- [ ] Documentation update

## Issues
- Resolves #NA
- OPIK-2856

### Manual Testing
- Tested time filtering on both `/threads` and `/threads/search` endpoints
- Verified correct filtering of threads within time bounds
- Confirmed validation errors for incomplete or invalid time parameters
- Verified thread model IDs are generated using earliest trace timestamp

### Automated Testing
All 10 tests passing:
- ✅ `whenTimeParametersProvided_thenIncludeThreadsWithinBounds` (stream & paginated)
- ✅ `whenTimeParametersInISO8601Format_thenReturnFilteredThreads` (stream & paginated)
- ✅ `whenTimeParametersProvided_thenExcludeThreadsOutsideBounds` (stream & paginated)
- ✅ `whenOnlyFromTimeProvided_thenReturnBadRequest` (stream & paginated)
- ✅ `whenFromTimeAfterToTime_thenReturnBadRequest` (stream & paginated)

Run tests with:
```bash
cd apps/opik-backend
mvn test -Dtest="FindTraceThreadsResourceTest\$FindTraceThreadsTimeFilteringTests"
```

## Documentation
- No documentation updates required (API follows existing pattern for traces/spans)

### CODE REVIEW

**Comment by andrescrz** [comment]
@thiagohora please address test failures.

**Comment by thiagohora** [comment]
> @thiagohora, please address test failures.

Sorry, I was on a call and didn't see that I broke it.

File: apps/opik-backend/src/main/java/com/comet/opik/api/ThreadTimestamps.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package com.comet.opik.api;
+
+import java.time.Instant;
+import java.util.UUID;
+
+public record ThreadTimestamps(UUID firstTraceId, Instant lastUpdatedAt) {
```
Comment (andrescrz) [inline apps/opik-backend/src/main/java/com/comet/opik/api/ThreadTimestamps.java:6]: Minor: add builder (with to builder) and use it as better option than constructor. Also add validations if needed, such as NonNull, as it's a public record.

File: apps/opik-backend/src/main/java/com/comet/opik/domain/IdGenerator.java [hunk 2]
```diff
@@ -13,6 +14,24 @@ public interface IdGenerator {
 
     UUID getTimeOrderedEpoch(long epochMilli);
 
+    default UUID generateId(Instant timestamp) {
```
Comment (andrescrz) [inline apps/opik-backend/src/main/java/com/comet/opik/domain/IdGenerator.java:17]: Minor: better move all these implementations to the `IdGeneratorImpl`. This isn't a public library, we control the usage an implementation within this service. So no need to keep backwards compatibility by providing default implementations.

I'd rather favour clean interfaces, without implementation details.

File: apps/opik-backend/src/main/java/com/comet/opik/domain/IdGenerator.java [hunk 3]
```diff
@@ -13,6 +14,24 @@ public interface IdGenerator {
...
+        return getTimeOrderedEpoch(timestamp.toEpochMilli());
+    }
+
+    /**
+     * Extracts the Unix epoch timestamp in milliseconds from a UUIDv7.
+     *
+     * @param uuid the UUIDv7 instance
+     * @return the extracted timestamp as Instant
+     */
+    static Instant extractTimestampFromUUIDv7(UUID uuid) {
```
Comment (andrescrz) [inline apps/opik-backend/src/main/java/com/comet/opik/domain/IdGenerator.java:27]: Minor: add validations to the implementation of all these new public methods.
